imageprocessing/src/main/java/com/example/imageprocessing/PreProcessing.java

```diff
@@ -0,0 +1,394 @@
+        //image in the right way
+        Matrix matrix = new Matrix();
+        matrix.preRotate((float)-degrees);
+
+        //Rotate the Bitmap and returns it
+        return Bitmap.createBitmap(original, 0, 0, width, height, matrix, true);
+
+    }
+
+
+    /**
+     * Performs the skew correction
+     * @param image The image that I want to rotate, must not be null
+     * @return The image corrected
+     * @author Thomas Porro(g1), Giovanni Fasan (g1), Oscar Garrido (g1)
+     */
+    private Bitmap editSkew(@NonNull Bitmap image){
+        double angle = computeSkew(image);
+        return rotateBitmap(image, angle);
+    }
+
+
+    /**
+     * Detect if the image is bright
+     * @param imageMat the image we want to detect the brightness
+     * @return IMAGE_IS_OK if image is neither too bright nor too dark,
+     *         IMAGE_IS_BRIGHT if image is too bright,
+     *         IMAGE_IS_DARK if image is too dark.
+     * @author Thomas Porro(g1), Giovanni Fasan(g1), Leonardo Pratesi(g1)
+     */
+    private BrightnessValue isBright(Mat imageMat){
+        //Converts the image into a matrix
+        Mat brightnessMat = new Mat();
+
+        //Changes the format of the matrix into a RGB one
+        Imgproc.cvtColor(imageMat, brightnessMat, Imgproc.COLOR_RGBA2RGB);
+
+        //Obtain 3 different matrix with the 3 elemental colors
+        List<Mat> color = new ArrayList<>();
+        Core.split(brightnessMat, color);
+
+        /*Each color is multiplied with his luminance.
+          The colors are in order RGB, so to access the che color I use the number 0, 1, 2 in order
+          For more informations see https://en.wikipedia.org/wiki/Relative_luminance*/
+        Mat redLuminance = new Mat();
+        Core.multiply(color.get(0), new Scalar(0.2126), redLuminance);
+        Mat greenLuminance= new Mat();
+        Core.multiply(color.get(1), new Scalar(0.7152), greenLuminance);
+        Mat blueLuminance = new Mat();
+        Core.multiply(color.get(2), new Scalar(0.0722), blueLuminance);
+
+        //Sums the matrix of the colors into a single one
+        Mat tempLuminance = new Mat();
+        Mat totalLuminance = new Mat();
+        Core.add(redLuminance , greenLuminance , tempLuminance); //Red + Green = Temp
+        Core.add(tempLuminance , blueLuminance , totalLuminance); //Temp + Blue = Luminance
+
+        //Calculate the sum of the values of all pixels
+        Scalar sum = Core.sumElems(totalLuminance);
+
+        //Determines the image's bit
+        int bit;
+        switch ( brightnessMat.depth() ) {
+            case CV_8U:  bit = 8; break;
+            case CV_8S:  bit = 8; break;
+            case CV_16U: bit = 16; break;
+            case CV_16S: bit = 16; break;
+            case CV_32S: bit = 32; break;
+            case CV_32F: bit = 32; break;
+            case CV_64F: bit = 64; break;
+            default: return BrightnessValue.IMAGE_IS_OK;
```

Comment:
BrightnessValue is an enum and BrightnessValue.IMAGE_IS_OK corresponds to 0. 
The cases contain all the possible bits of an image, right? When I don't get one of them I return IMAGE_IS_OK but if there is an error in the image and opencv is not able to read its depth this is not correct because I can't know if the brightness of the image is ok.

Reply:
We understood that we'll always have 8 bit images so we are removing the switch